src/w3iProxy/pushProviders/internalPushProvider.ts

```diff
@@ -32,16 +38,52 @@ export default class InternalPushProvider implements W3iPushProvider {
     return `An initialized PushClient is required for method: [${method}].`
   }
 
-  public isListeningToMethodFromPostMessage() {
-    return false
+  public isListeningToMethodFromPostMessage(method: string) {
+    return this.methodsListenedTo.includes(method)
   }
 
-  public handleMessage() {
-    throw new Error(`${this.providerName} does not support listening to external messages`)
+  public handleMessage(request: JsonRpcRequest<unknown>) {
+    switch (request.method) {
+      case 'push_signature_delivered':
+        this.emitter.emit('push_signature_delivered', request.params)
+        break
+      default:
+        throw new Error(`Method ${request.method} unsupported by provider ${this.providerName}`)
+    }
   }
 
   // ------------------- Method-forwarding for PushWalletClient -------------------
 
+  public async enableSync(params: { account: string }) {
+    if (!this.pushClient) {
+      throw new Error(this.formatClientRelatedError('approve'))
+    }
+
+    return this.pushClient.enableSync({
+      ...params,
+      onSign: async message => {
+        this.emitter.emit('push_signature_requested', { message })
+
+        return new Promise(resolve => {
+          setInterval(() => {
```

Comment:
Looks like no `clearInterval` being called for this before `resolve`. My suspicion is this will keep running beyond the `resolve` without it.

Reply:
More generally: why do we have to poll here, are we missing an event or some other affordance to detect something more cleanly in syncClient?

Reply:
It would add more complexity, due to the fact that `chatClient.register` and `pushClient.enableSync` are fired after one another, emitting both `chat_signature_requested` and `push_signature_requested`, so the other solution is to listen to multiple events and orchestrate signing.

I went with the simpler approach, good call on `clearInterval` though. Slipped my mind